docs(aws sqs): add AWS SQS OOTB connector doc (#1209)
camunda/camunda-platform-docs: docs/components/integration-framework/connectors/out-of-the-box-connectors/aws-sqs.md

---
id: aws-sqs
title: AWS Simple Queue Service Connector
description: Send messages to AWS Simple Queue Service from your BPMN process.
---

The AWS SQS Connector allows you to connect your BPMN service with the Amazon Web Service's [Simple Queue Service](https://aws.amazon.com/sqs/).

## Prerequisites

> To use an **AWS SQS Connector**, you need to have an SQS Queue and IAM key and secret pair with `sqs:SendMessage` policy relative to your SQS. See the [SQS Developer Guide](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-basic-examples-of-iam-policies.html) to learn more.

Reviewer: Have you thought about adding something like

```
... and configured as secret in your cluster. Follow [these steps in the appendix](...) to learn how to set this up.
```

+ small appendix section `Store SQS key and secret as secrets in your cluster`

==> Example https://docs.camunda.io/docs/next/components/integration-framework/connectors/out-of-the-box-connectors/slack/#appendix

Author: Done

1. Set relevant IAM key and secret pair in the **Authentication** section. For example, `secrets.MY_AWS_ACCESS_KEY`. The value can be plain text, but this is not recommended due to security concerns.
2. In the **Queue Properties** section, set the URL of your SQS queue as well as its region.
3. In the **Input message data** section, fill the field **Message body** with the data you would like to submit to the queue. The field requires FEEL input.
4. (Optional) In the **Input message data** section, fill the field **Message attributes** to set optional message metadata. This field requires FEEL input. See the relevant [appendix](#what-are-the-message-attributes-and-how-can-i-set-them) section to find out more about this field.

## AWS SQS Connector response

The **AWS SQS Connector** returns the SQS message identifier of a newly created message.
The response contains a `messageId` variable.

You can use an output mapping to map the response:

1. Use **Result Variable** to store the response in a process variable. For example, `myResultVariable`.
2. Use **Result Expression** to map fields from the response into process variables. For example:

```
= {

>   "myCreatedMessage": response.messageId

Reviewer: ```suggestion
  "createdMessageID": response.messageId
```

---
id: aws-sqs

> title: AWS Simple Queue Service Connector

Reviewer 2: ```suggestion
title: Amazon Simple Queue Service Connector
```

> description: Send messages to AWS Simple Queue Service from your BPMN process.

Reviewer 2: ```suggestion
description: Send messages to Amazon Simple Queue Service from your BPMN process.
```

---

> The AWS SQS Connector allows you to connect your BPMN service with the Amazon Web Service's [Simple Queue Service](https://aws.amazon.com/sqs/).

Reviewer 2: Minor tweak. Update text to: "The AWS SQS Connector allows you to connect your BPMN service with Amazon Web Service's [Simple Queue Service](https://aws.amazon.com/sqs/)."

> The AWS SQS Connector allows you to connect your BPMN service with the Amazon Web Service's [Simple Queue Service](https://aws.amazon.com/sqs/).

Author: ```suggestion
The Amazon SQS Connector allows you to connect your BPMN service with the Amazon Web Service's [Simple Queue Service](https://aws.amazon.com/sqs/).
```

> The AWS SQS Connector allows you to connect your BPMN service with the Amazon Web Service's [Simple Queue Service](https://aws.amazon.com/sqs/).

Reviewer 2: ```suggestion
The Amazon SQS Connector allows you to connect your BPMN service with Amazon Web Service's [Simple Queue Service](https://aws.amazon.com/sqs/).
```

## Prerequisites

> To use an **AWS SQS Connector**, you need to have an SQS Queue and IAM key and secret pair with `sqs:SendMessage` policy relative to your SQS.

Reviewer 2: Update text to "To use the **AWS SQS Connector**, you need..."

> To use an **AWS SQS Connector**, you need to have an SQS Queue and IAM key and secret pair with `sqs:SendMessage` policy relative to your SQS.

Reviewer 2: ```suggestion
To use the **Amazon SQS Connector**, you need to have an SQS Queue and IAM key and secret pair with the `sqs:SendMessage` policy relative to your SQS.
```

> To use the **Amazon SQS Connector**, you need to have an SQS Queue and IAM key and secret pair with the `sqs:SendMessage` policy relative to your SQS.

Author: ```suggestion
To use the **Amazon SQS Connector**, you need to have an SQS Queue and IAM key and secret pair with the `lambda:InvokeFunction` policy relative to your SQS.
```

It is highly recommended not to expose your AWS IAM credentials as plain text but rather use Camunda Secrets. See an [appendix entry](#how-to-store-aws-iam-secrets-for-my-sqs-connector) to find out more.
See the [SQS Developer Guide](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-basic-examples-of-iam-policies.html) to learn more.

> ## Create an AWS SQS Connector task

Reviewer 2: ```suggestion
## Create an Amazon SQS Connector task
```

> To use an **AWS SQS Connector** in your process, either change the type of existing task by clicking on it and using the wrench-shaped **Change type** context menu icon, or create a new Connector task by using the **Append Connector** context menu. Follow our [guide on using Connectors](../use-connectors.md) to learn more.

Reviewer 2: ```suggestion
To use the **Amazon SQS Connector** in your process, either change the type of existing task by clicking on it and using the wrench-shaped **Change type** context menu icon, or create a new Connector task by using the **Append Connector** context menu. Follow our [guide on using Connectors](../use-connectors.md) to learn more.
```

> ## Make your AWS SQS Connector for sending messages executable

Reviewer 2: ```suggestion
## Make your Amazon SQS Connector for sending messages executable
```

![AWS SQS Filled](../img/connectors-aws-sqs-filled.png)

1. Set relevant IAM key and secret pair in the **Authentication** section. For example, `secrets.MY_AWS_ACCESS_KEY`. The value can be plain text, but this is not recommended due to security concerns.
2. In the **Queue Properties** section, set the URL of your SQS queue as well as its region.
3. In the **Input message data** section, fill the field **Message body** with the data you would like to submit to the queue. The field requires FEEL input.
4. (Optional) In the **Input message data** section, fill the field **Message attributes** to set optional message metadata. This field requires FEEL input. See the relevant [appendix](#what-are-the-message-attributes-and-how-can-i-set-them) section to find out more about this field.

> ## AWS SQS Connector response

Reviewer 2: ```suggestion
## Amazon SQS Connector response
```

> The **AWS SQS Connector** returns the SQS message identifier of a newly created message.

Reviewer 2: ```suggestion
The **Amazon SQS Connector** returns the SQS message identifier of a newly created message.
```

The response contains a `messageId` variable.

You can use an output mapping to map the response:

1. Use **Result Variable** to store the response in a process variable. For example, `myResultVariable`.
2. Use **Result Expression** to map fields from the response into process variables. For example:

```
= {
  "createdMessageID": response.messageId
}
```

## Appendix & FAQ

### What are the message attributes and how can I set them?

Amazon SQS lets you include structured metadata (such as timestamps, geospatial data, signatures, and identifiers) with messages using message attributes.

> The **AWS SQS Connector** allows you to include non-binary message attributes in the section **Input message data**. The message attribute value has to be composed to be compliant with AWS SQS [message attribute data format](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-message-metadata.html#sqs-message-attributes).

Reviewer 2: ```suggestion
The **Amazon SQS Connector** allows you to include non-binary message attributes in the section **Input message data**. The message attribute value has to be composed to be compliant with Amazon SQS [message attribute data format](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-message-metadata.html#sqs-message-attributes).
```

Example of a valid message attribute as a FEEL value:

```
= {
  "timestamp":{
    "StringValue":today(),
    "DataType":"String"
  },
  "messageSubmittedBy":{
    "StringValue":"user12345",
    "DataType":"String"
  }
}
```

> ![AWS SQS Message Attributes](../img/connectors-aws-sqs-message-attributes.png)

Reviewer 2: ```suggestion
![Amazon SQS Message Attributes](../img/connectors-aws-sqs-message-attributes.png)
```

> ### How to store AWS IAM Secrets for my SQS Connector?

Reviewer 2: ```suggestion
### How do I store AWS IAM Secrets for my SQS Connector?
```
